### DISCUSSION TITLE
LaTeX / MathJax rendering bugs

### DISCUSSION THREAD

**Discussion by avm19** [community/community#18897] [Other Feature Feedback, Questions, & Ideas] [answered]
Just a week or two ago this text would have rendered correctly in discussions, issues, PR messages etc:
`Bla $\sum_j j$ but $\sum_j$`
At the time of writing, it looks like that:
![image](https://user-images.githubusercontent.com/52547519/174463433-6128e7f8-ee4a-4410-96d1-908ccdaaebee.png)
Here is how it is being rendered when you are reading this message:
Bla $\sum_j j$ but $\sum_j$
Here is what it supposed to look like (rendered at stackexchange.com):
![image](https://user-images.githubusercontent.com/52547519/174463667-f7bbcab8-e685-438a-85b8-4d1c44f1ba6e.png)

Here are related discussions **3-4 weeks ago**: 
https://github.com/github-community/community/discussions/16939
https://github.com/github-community/community/discussions/17384

I emphasise that _certainly_ **two weeks ago** (and _probably_ also 1 week ago) this bug did not occur (in case you are interested, here is [the message](https://github.com/scikit-learn/scikit-learn/pull/22574#discussion_r889601021), which I remember to have looked fine). You must have pushed some buggy changes to the rendering algorithm within this time period.

------------

**Update**

This issue is known to the support, here is the answer I received after submitting a bug report through the link kindly suggested in the messages below.

> Hi avm19,
> 
> Thanks for reaching out to GitHub Support about problems rendering multiple inline math statements in a line. We have received some reports of this and our team is currently working on a fix for this. I have added your report to the internal issue where this is being tracked and we will follow-up with you when this has been fixed.
> 
> Note that while this support ticket may automatically close, that the internal issue will remain open and under investigation.
> 
> Thanks,

**Update-2**. @nschloe suggested the following workaround: 
* for inline formulae, use `` $` <your_LaTeX> `$ ``
* for displayed formulae, use `` ```math <your_LaTeX> ``` ``

Test for the minimal example `${x}_{a} {y}_{b}$`, which does not render correctly: ${x}_{a} {y}_{b}$
* [+] inline dollar-backtick-escaped: $`{x}_{a} {y}_{b}`$
* [+] displayed math code: 
```math
{x}_{a} {y}_{b}
```

**Comment by DLu** [comment]
I have also encountered this. [Here is my example](https://gist.github.com/DLu/2da201d4135e0d74f552b9b729fa024e)

**Comment by ylamidon** [comment]
Same problem. <a href="https://gist.github.com/ylamidon/5383c6b1d3923901695927340de6ed91">My example</a>

A workaround is to go the next line before $\delta$. <a href="https://gist.github.com/ylamidon/96ad0a13da1a1bc286e37c2d837c8776">A workaround</a>

**Reply by avm19** [reply]
This is very clever to use linebreaks to break Markdown contexts that overarch multiple LaTeX formulas!

Unfortunately, it works well only in markdown file renders, but not in messages (like this one), since single linebreaks are not ignored here.

**Comment by MathiasReker** [comment]
Hello. You can report the bug here: https://support.github.com/contact/bug-report

**Reply by avm19** [reply]
Thanks. I couldn't find this link at all when navigating the website.

**Comment by avm19** [comment]
This issue is known to the support. I included their response in the head message.

**Comment by cloudy-sfu** [comment]
Here is my example: https://github.com/github-community/community/discussions/20154

**Comment by DLu** [comment]
I believe this is fixed now 

Bla $\sum_j j$ but $\sum_j$

**Reply by avm19** [reply]
True.

Update: see [mbaudin47's comment](https://github.com/orgs/community/discussions/18897#discussioncomment-6075708).

**Comment by mbaudin47** [comment]
I do not think that this is fixed. 

The equation $\boldsymbol{x}_0$ is OK, 

but $\{\boldsymbol{x}_1, ..., \boldsymbol{x}_{i-1}, \boldsymbol{x}_{i+1}, ..., \boldsymbol{x}_n\}$ is not.

**Reply by avm19** [reply]
True. `${x}_{a} {y}_{b}$` would be close to the minimal example that fails. At the same time, `${x}_a {y}_b$` works: ${x}_a {y}_b$.  A temporary dirty fix would be escaping underscores, as in `${x}\_{a} {y}\_{b}$` (${x}\_{a} {y}\_{b}$), which is wrong because one should not and would not escape those underscores in LaTeX.

**Reply by nschloe** [reply]
Recommendation: Avoid $-math, always use

````markdown
$`a+b`$

```math
x+y
```
````

This avoids most issues, such as this one:
```markdown
$`{\boldsymbol{x}_1, ..., \boldsymbol{x}_{i-1}, \boldsymbol{x}_{i+1}, ..., \boldsymbol{x}_n}`$
```

> $`{\boldsymbol{x}_1, ..., \boldsymbol{x}_{i-1}, \boldsymbol{x}_{i+1}, ..., \boldsymbol{x}_n}`$

**Reply by avm19** [reply]
@nschloe Nice! Although you can't avoid $-math if you want an inline formula (can you?), you can add backticks to mess with GitHub's markdown translator, so that it produces the desired result. That is,

- for displayed formulae, use ` ```math <your_LaTeX> ``` `
- for inline formulae, use `` $` <your_LaTeX> `$ ``

Test:
- [-] minimal example of the bug (doesn't render right): ${x}_{a} {y}_{b}$
- [+] inline dollar-backtick-escaped: $`{x}_{a} {y}_{b}`$
- [+] displayed math code: 
```math
 {x}_{a} {y}_{b}
 ```
 
 P.S. I don't know what `` ` `` is used in LaTeX, so this trick should be okay.

**Reply by nschloe** [reply, marked as answer]
> $-math if you want an inline formula (can you?), you can add backticks 

That's what I mean: `$`-math is without backticks, <code>$`</code>-math with. Always use the latter. (I consider it a mistake that GitHub ever allowed using $-math, it complicates things a _lot_.)

PS: See https://github.com/nschloe/github-math-bugs/ for a list of current math bugs.

**Comment by ndrean** [comment]
Is there an update on this one? Seems like nothing is rendering at all.

**Comment by JoWestermann** [comment]
I experience the same issue. Not even very simple expressions are rendered correctly, e.g. `Let $i=0$.` which is rendered as:
![image](https://github.com/user-attachments/assets/edd2b898-afc5-48d3-8fcf-93d9780edf18)
The output is the same no matter if I use $ or  $`.

I would like to emphasize that using math notation in markdown Readme files is a highly important feature for scientific applications. I had just migrated a project from gitlab to github, only to find out that the Readme file has become completely unreadable. I hope that this will be fixed soon!
